packages/backend/src/tokens/request.ts

```diff
@@ -23,6 +23,31 @@ import {
 } from './interstitialRule';
 import type { VerifyTokenOptions } from './verify';
 
+export interface RequestAdapter {
+  /**
+   * The headers method returns the value of the specified header. Keys given to this method are
+   * normalized to lowercase, as they exist in `packages/backend/src/constants.ts`.
+   * Implementing this method requires the user to manipulate the header keys accordingly.
+   *
+   * @param key The header name to retrieve
+   */
+  headers(key: string): string | undefined;
```

Comment:
I'm somewhat concerned with the proposed API.
1. Why did we use our own interface for `headers` here? Is there a reason we cannot use [Headers](https://developer.mozilla.org/en-US/docs/Web/API/Headers)?
2. The naming confuses me. I'd expect `headers` to be an object/instance, not a function that returns a string. I'd expect something along the lines of `headers.get` (see point 1)
3. How can we write a header? This is necessary for the loop detection mechanism

Reply:
I thought we'd be doing the following:
- clerk/backend polyfills fetch and Headers for all runtimes
- clerk/backend exposes a util to convert an object into a Headers instance 
- clerk/backend exposes an interface requiring at least a `headers` instance to be present.

Reply:
@nikosdouvlis 
> Why did we use our own interface for headers here? Is there a reason we cannot use [Headers](https://developer.mozilla.org/en-US/docs/Web/API/Headers)?

We want to give full control to each framework. `@clerk/backend` package only passes a header key and the implementation is totally up to each framework.

>The naming confuses me. I'd expect headers to be an object/instance, not a function that returns a string. I'd expect something along the lines of headers.get (see point 1)

I agree with the naming, we could rename to something like `getHeader`.

>How can we write a header? This is necessary for the loop detection mechanism

I think it's not. This is an adapter on how the `@clerk/backend` will have access to headers to call `authenticateRequest`. Writing on the headers is the framework's job. Am I missing a use case?

Reply:
`RequestAdapter` can expect a `Headers` instance instead of this proprietary `headers` method. Each framework will have full control on how to transform data from its request to a `Headers` instance.

Reply:
Writing headers is needed in order to get the loop detection mechanism work for every framework without reimplementing the logic.